### ISSUE TITLE
{Freeway-Bug}[Settings>Default terminal application]: Options present for the 'Default terminal application' combo box are getting truncated at 200% zoom.

### ISSUE THREAD

**Issue by Saiteja341** [microsoft/terminal#15149]
### Windows Terminal version

1.18.902.0

### Windows build number

23031.1000

### Other Software

Test environment:
OS: Version 22H2 (OS build 23031.1000)
Application: Windows Terminal
Screen reader: Narrator

### Steps to reproduce

Pre-requisite:
Apply the resize using following path.
Settings>Accessibility>Display>Display resolution>Change display resolution to '1280 x 720>Keep changes>Drag the 'Text size' slider to '200%'>Apply>Re-open the application.

Repro steps:

1. Open the 'windows terminal' application.
2. Log in with valid credentials.
3. Follow the above-mentioned pre-requisite step.
4. Open settings using 'Ctrl + ,'.
5. Navigate to 'Start up' list-item & select it.
6. Navigate to the 'Default terminal application' combo box & activate it.
7. Navigate among the options present in the expanded combo box.
8. Observe the issue.

User experience:
Users with low vision rely on zooming features for navigation will feel difficulty in understanding the purpose of options present for the combo box, if options are not visible properly at 200% zoom.

Guideline reference:
https://www.w3.org/TR/WCAG21/#resize-text

Attachment:
[Options present for the combo box are not visible properly on re-sizing..zip](https://github.com/microsoft/terminal/files/11189870/Options.present.for.the.combo.box.are.not.visible.properly.on.re-sizing.zip)

### Expected Behavior

In 200% zoom, options present for the expanded 'Default terminal application' combo box should be visible completely without any overlapping, as they visible in 100%zoom.

### Actual Behavior

Options present for the 'Default terminal application' combo box are getting truncated & overlapping at 200% zoom.
